_posts/en/newsletters/2020-09-23-newsletter.md

```diff
@@ -0,0 +1,264 @@
+  pinning][topic transaction pinning], can be used to attack contract
+  protocols.  This has lead to partial solutions such as [anchor
+  outputs][topic anchor outputs] that work with specific protocols (e.g.
+  two-party LN channels) but aren't easy to generalize and so are
+  somewhat unsatisfactory.
+
+    At the consensus layer, Rubin proposes allowing transactions to
+    optionally contain a special final output that commits to the txids
+    of one or more other unconfirmed transactions.  The transaction with
+    the special output, a *sponsor* transaction, may only be included in
+    a valid block if every one of the transactions it *sponsored* is
+    also included in that same block.  This means a miner who wants the
+    high feerate from a sponsor transaction will be incentivized to also
+    confirm the sponsored transactions even if those transactions pay a
+    low feerate.  Beyond their special sponsorship output, sponsor
+    transactions are normal Bitcoin transactions.
+
+    At the policy layer that controls mempool acceptance and transaction
+    relay, Rubin suggests a set of simple rules that allow anyone to
+    sponsor any transaction currently in a mempool or to replace an
+    existing sponsor transaction with higher feerate alternative that
+    makes the same commitment.  The goal is to ensure that, as long as a
+    sponsor transaction follows the rules and pays a high enough
+    feerate, it will propagate across the network without running afoul
+    of pinning or other attacks.
+
+    Rubin's proposal comes with a [reference implementation][rubin
+    refimpl] and discussion of design tradeoffs and forward
+    compatibility.  As of this writing, comments on the list have been
+    positive but non-committal---the proposal needs to not only provide
+    satisfactory solutions to existing problems but prove that it is
+    both absent of any significant new problems and that it will attract
+    the support needed to introduce a change to Bitcoin's consensus
+    rules.
+
+- **Research into conflicts between timelocks and heightlocks:** A
+  [post][blockstream post] to the Blockstream engineering blog describes
+  an interaction between different types of Bitcoin time locks
+  (timelocks) and block-height locks (heightlocks).  All transactions
+  since Bitcoin's initial release have had an `nLockTime` field.  A soft
+  fork activating at block 31,000 (December 2009) began
+  [comparing][time-height-lock fork] this field against a block header's
+  explicit nTime field and its implicit height (number of blocks since
+  the Genesis Block).  Using that comparison, a block is only valid if
+  every one of its transactions follows these two
+  rules:[^final-sequence]
+
+    - If a transaction's nLockTime is below 500 million, it must also be
+      below the block's height.
+
+    - If a transaction's nLockTime is equal to or above 500 million, it
+      must be below the block header's nTime (in [epoch time][]).
```

Comment:
Nowadays it looks like it should be s/nTime/`nBlockTime`/ here and in lines 71 and 90 per https://github.com/bitcoin/bitcoin/commit/cc0b4c3b62367a2aebe5fc1f4d0ed4b97e9c2ac9#diff-23cfe05393c8433e384d2c385f06ab93R404-R413 and `git grep IsFinalTx`

(Edit: discussed on IRC, so feel free to ignore unless you feel it's worth updating here.)